tests/unit/plugins/module_utils/test_xml_utils.py

```diff
@@ -415,6 +415,21 @@ def test_elements_equal_without_children_whitespace_matches():
     assert xml_utils.elements_equal(e1, e2)
 
 
+def test_elements_equal_str_num_and_int():
+    """
+    Test that elements_equal function considers a string number equal to an integer number
+
+    eg:
+    <test/> == <test>   </test>
+    """
+    e1 = Element("test")
+    e1.text = 1
+    e2 = Element("test")
+    e2.text = "1"
+
+    assert xml_utils.elements_equal(e1, e2)
+
+
 def test_elements_equal_without_children_none_and_1():
     """
     Tests elements_equal function matches a boolean flag "1" the same as an empty element.
```

Comment:
you might consider updating the description

Reply:
IMO this test doc describes the test case. Do you have a specific suggestion in mind?